### DISCUSSION TITLE
Incompatible with psychedelic shaders

### DISCUSSION THREAD

**Discussion by VexianEmpire** [iPortalTeam/ImmersivePortalsMod#1607] [Non-Compatibility Issue Report]
![image](https://user-images.githubusercontent.com/64798986/81444303-885bbc00-913c-11ea-807b-d1d8dc33aa48.png)

The portals are rendered as if the shaders are not there

![image](https://user-images.githubusercontent.com/64798986/81444677-28194a00-913d-11ea-92bd-387b69ca0cc2.png)

It only looks correct when you get close

![image](https://user-images.githubusercontent.com/64798986/81444717-37989300-913d-11ea-82b2-34d17a2d1f32.png)

![image](https://user-images.githubusercontent.com/64798986/81446093-cefee580-913f-11ea-9625-9618a12fe023.png)

![image](https://user-images.githubusercontent.com/64798986/81446193-04a3ce80-9140-11ea-8ee7-f737a184122e.png)

**Comment by VexianEmpire** [comment]
http://shadersmods.com/category/psychedelic-shaders/

**Comment by LoganDark** [comment]
Immersive Portals' renderers were not designed to be used with shaders that actually distort space. I believe it simply uses the portals' shape rather than going through vertex shaders and what have you. I believe this is WAI

**Comment by LoganDark** [comment]
Also, is shadersmods.com like CurseForge or does it need some https://stopmodreposts.org/ ?

**Comment by VexianEmpire** [comment]
I don't really know, I just used it as an example, but here's an image

![image](https://user-images.githubusercontent.com/64798986/81447183-cb6c5e00-9141-11ea-9b67-85c05047e021.png)

**Comment by LoganDark** [comment]
I actually checked it out and it seems like they're just storing direct links to shader mods.

**Comment by VexianEmpire** [comment]
I don't think it'd be that hard to implement as the portals could warp just on each grid point as these blocks do:
![image](https://user-images.githubusercontent.com/64798986/81453096-49cefd00-914e-11ea-9117-0144cf277fc0.png)
![image](https://user-images.githubusercontent.com/64798986/81453127-59e6dc80-914e-11ea-8e58-6c31493f17d1.png)
You wouldn't have to add any points, just move them. (If we're dealing with a large portal, you could add a few points to prevent it from being a straight line instead of curved as the shaders intended)

**Comment by qouteall** [comment]
Supporting psychedelic shaders is very hard and not worth the effort.

**Comment by jbritain** [comment]
Hi, I assume support for this is still unlikely to be added? I am working on a project involving custom vertex shaders to distort the world and immersive portals, but none of the available optifine gbuffers vertex shaders have any effect on the portal edges.
